### ISSUE TITLE
Tests time out: System.Net.Sockets.Tests.DnsEndPointTest - mostly on SLES.15 in release/6.0 branches

### ISSUE THREAD

**Issue by karelz** [dotnet/runtime#57929]
Failures:
- `Socket_ConnectAsyncDnsEndPoint_HostNotFound`
  - 6/24-8/23 (incl. PRs): **241** -- each day at least one except 8/4 and 8/7.
    - Failures are on release/6.0* branches (preview3+) -- only 1 failure on 7/16 is in main branch, no failure on any PR
    - OS failures:
      - SLES.15.Amd64.Open - **236x**
      - Windows.Nano.1809.Amd64.Open - 2x (1 occurrence on main branch - 7/16, 7/18)
      - Windows.10.Amd64.Server19H1.Open - 1x (7/30)
      - Windows.7.Amd64.Open - 1x (8/12)
      - Windows.10.Amd64.ServerRS5.Open - 1x (8/16)
  - 8/24-8/29 (incl. PRs): **15** - all release/6.0*, official builds, SLES.15.Amd64.Open
  - 8/30-9/5 (incl. PRs): **20** - all release/6.0*, 2x PR #58336, SLES.15.Amd64.Open
- `Socket_StaticConnectAsync_HostNotFound`
  - 6/24-8/23 (incl. PRs): **95** -- each day at least one except 7/7, 7/18, 8/5, 8/7-8/10, 8/12, 8/14-8/15, 8/17-8/18, 8/20.
    - All failures are on release/6.0* branches (preview3+)
    - OS failures:
      - SLES.15.Amd64.Open - **93x**
      - Windows.Nano.1809.Amd64.Open - 1x (7/17)
      - Windows.10.Amd64.Server19H1.Open - 1x (7/22)
  - 8/24-8/29 (incl. PRs): **9** -- all release/6.0*, official builds, SLES.15.Amd64.Open
  - 8/30-9/5 (incl. PRs): **12** -- 1x main=7.0 on Windows.10.Amd64.Server19H1.Open, otherwise release/6.0* on SLES.15.Amd64.Open, 2x PR #58336
- `Socket_StaticConnectAsync_Success` -- **main branch failures tracked in #21924**
  - 6/24-8/23 (incl. PRs): **21**
    - Mix of branches and OSs
      - 4x main + 1x PR + 16x release/6.0-preview* branches
      - OSs:
        - SLES.15.Amd64.Open - 4x
        - Debian.9.Arm32.Open - 4x
        - OSX.1015.Amd64.Open - 2x (incl. 1x Mono)
        - Ubuntu.1804.ArmArch.Open - 2x
        - Ubuntu.1910.Amd64.Open - 2x
        - Fedora.32.Amd64.Open - 3x
        - OSX.1013.Amd64.Open - 1x
        - Windows.10.Amd64.ServerRS5.Open - 1x
        - Debian.9.Amd64.Open - 1x
        - Windows.10.Amd64.Server19H1.Open - 1x
  - 8/24-8/29 (incl. PRs): **2** -- all release/6.0*, official builds -- 1x SLES.15.Amd64.Open, 1x Debian.9.Amd64.Open
  - 8/30-9/5 (incl. PRs): **1** -- main=7.0 official run on Ubuntu.1804.ArmArch.Open

Kusto query:
```kusto
let failedTests = (methodName : string, includePR : bool, messageSubstr: string) {
cluster('engsrvprod.kusto.windows.net').database('engineeringdata').TestResults
| where Type startswith "System.Net" 
    and Result == 'Fail'
| where Method == methodName
  or Method == 'Socket_StaticConnectAsync_Success'
  or Method == 'Socket_StaticConnectAsync_HostNotFound'
  or Method == 'Socket_ConnectAsyncDnsEndPoint_HostNotFound'
| where Message contains messageSubstr
| distinct JobId, WorkItemId, Message, Exception, StackTrace, Method, Type //, Arguments
| join kind=inner (cluster('engsrvprod.kusto.windows.net').database('engineeringdata').Jobs
    | where ((Branch == 'refs/heads/main') or (Branch == 'refs/heads/master') or (includePR and (Source startswith "pr/")) or (Branch startswith 'refs/heads/release/6.0'))
    | where Type startswith "test/functional/cli/"
        and not(Properties contains "runtime-staging")
    | summarize arg_max(Finished, Properties, Type, Branch, Source, Started, QueueName) by JobId
| project-rename JobType = Type) on JobId
};
failedTests('', true, '');
```

Failure:
```
Timed out while waiting for connection
Expected: True
Actual:   False

   at System.Net.Sockets.Tests.DnsEndPointTest.Socket_ConnectAsyncDnsEndPoint_HostNotFound() in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/DnsEndPointTest.cs:line 275
```

**Comment by karelz** [comment]
@antonfirsov any idea why only this test fails?
And 99% failures are on release/6.0 branches on SLES 15

**Comment by antonfirsov** [comment]
Async NameResolution is async-over-sync on Unix, queuing work on the ThreadPool in parallel with other socket tests. One explanation can be that the 10seconds timeout is too strict in some cases. Another one is that resolution of invalid names fails too slowly on these SLES machines. The two can be related.

I wonder if there is anything special to the release test runs in comparison to the main runs (images? machines? some sort of helix config?). Is the fix for #48751 applicable for these machines?  /cc @MattGal

**Comment by karelz** [comment]
@antonfirsov FYI -- 3rd test is similar failure (added above), but not unique to SLES and also reproduces more on main branch (5x).

**Comment by MattGal** [comment]
> Async NameResolution is async-over-sync on Unix, queuing work on the ThreadPool in parallel with other socket tests. One explanation can be that the 10seconds timeout is too strict in some cases. Another one is that resolution of invalid names fails too slowly on these SLES machines. The two can be related.
> 
> I wonder if there is anything special to the release test runs in comparison to the main runs (images? machines? some sort of helix config?). Is the fix for #48751 applicable for these test runs? /cc @MattGal

We had a chat and I did note that .svc is using D2_v4 (Intel-based)  while all the other SLES.15 queues are using D2a_v4 (AMD-EPYC based) processors.  Nothing else is particularly interesting, and the same test has passed on .svc in the past.  Anton is going to investigate, depending on the outcome it may be worth just upping the timeout.

**Comment by karelz** [comment]
Reopening to track backport to 6.0

**Comment by karelz** [comment]
Fixed in 6.0 RC2 in PR #58617.
Fixed in 7.0 in PR #58129.
